### PR TITLE
order wh uploads by upload with oldest event

### PR DESCRIPTION
**Pull request by psrikanth88**
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: warehouse/warehouse.go [hunk 1]
```diff
@@ -640,6 +693,7 @@ func (wh *HandleT) mainLoop() {
 		}
 
 		warehouses := wh.warehouses
+		wh.sortWarehousesByLastEventAt(warehouses)
```
Comment (SumanthPuram) [inline warehouse/warehouse.go:696]: Instead of sorting it here, let's use this logic to create upload jobs in the uploadJobAllocator?

**Review by SumanthPuram — COMMENTED** [review]
Refactor this?

File: warehouse/utils/utils.go [hunk 2]
```diff
@@ -96,6 +96,7 @@ type WarehouseT struct {
 	Namespace   string
 	Type        string
 	Identifier  string
+	// LastUploadedEventTime time.Time
```
Comment (SumanthPuram) [inline warehouse/utils/utils.go:99]: Remove.

File: warehouse/warehouse.go [hunk 3]
```diff
@@ -622,16 +624,67 @@ func (wh *HandleT) processJobs(warehouse warehouseutils.WarehouseT) (numJobs int
...
-	wh.enqueueUploadJobs(uploadJobs, warehouse)
+	wh.uploadJobsQ <- uploadJobs
 	enqueuedJobs = true
 	return len(uploadJobs), nil
 }
 
+func (wh *HandleT) sortWarehousesByLastEventAt(warehouses []warehouseutils.WarehouseT) {
+	sqlStatement := fmt.Sprintf(`
+		SELECT
+			concat(source_id, '_', destination_id) as wh_identifier, last_event_at
```
[the remark this answers is not in the document]
Comment (SumanthPuram) [reply warehouse/warehouse.go:640]: Don't need to pass warehouses here. It's already present in `wh`

Also rename the func to the following. WDYT?

wh.updateWarehousesProcessingOrder()

**Review by SumanthPuram — APPROVED** [review]

**Review by gane5hvarma — APPROVED** [review]

**Pull request merged**